x-pack/plugins/monitoring/server/lib/alerts/fetch_nodes_from_cluster_stats.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { get } from 'lodash';
+import { AlertCluster, AlertClusterStatsNodes } from '../../../common/types/alerts';
+
+interface ClusterStateNodesESResponse {
+  [nodeUuid: string]: {
+    name: string;
+    ephemeral_id: string;
+  };
+}
+
+function formatNode(nodes: ClusterStateNodesESResponse) {
+  return Object.keys(nodes).map((nodeUuid) => {
+    return {
+      nodeUuid,
+      nodeEphemeralId: nodes[nodeUuid].ephemeral_id,
+      nodeName: nodes[nodeUuid].name,
+    };
+  });
+}
+
+export async function fetchNodesFromClusterStats(
+  callCluster: any,
+  clusters: AlertCluster[],
+  index: string
+): Promise<AlertClusterStatsNodes[]> {
+  return await Promise.all(
```

Comment:
Can this maybe be a single query, like the other alerts?

Reply:
Yup, done